### PR TITLE
Move VPN Android app to be channel of main VPN app

### PR DESCRIPTION
**Pull request by sean-rose**
Both are from the same [mozilla-mobile/mozilla-vpn-client](https://github.com/mozilla-mobile/mozilla-vpn-client) repo, both use the same metrics and ping files (though I understand the different Glean SDKs used may cause minor differences), and the VPN team would like unified views of all the VPN app Glean data in BigQuery & Looker ([DENG-210](https://mozilla-hub.atlassian.net/browse/DENG-210)).

Based on a dry run, the only consequential result of this change is in the v2 API the Android app's `app_name` would change from `mozilla_vpn_android` to `mozilla_vpn` (its `app_id` would remain the same).

### CODE REVIEW

File: repositories.yaml
Comment (sean-rose) [inline repositories.yaml:808]: Adding Bea here because she was in the notification list for the Android app, but now that she has moved to a different role, maybe this should actually be removed altogether?

And/or should I add myself?  (I'm unsure what kind of notifications these would be)

Comment (sean-rose) [reply repositories.yaml:808]: @brizental do you still want these notifications?

**Review by relud — APPROVED** [review]

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
@badboy can you take a look at this before it gets merged?

> Both are from the same [mozilla-mobile/mozilla-vpn-client](https://github.com/mozilla-mobile/mozilla-vpn-client) repo, both use the same metrics and ping files (though I understand the different Glean SDKs used may cause minor differences), and the VPN team would like unified views of all the VPN app Glean data in BigQuery & Looker ([DENG-210](https://mozilla-hub.atlassian.net/browse/DENG-210)).

Will this change automatically generate an unified view?

> Based on a dry run, the only consequential result of this change is in the v2 API the Android app's `app_name` would change from `mozilla_vpn_android` to `mozilla_vpn` (its `app_id` would remain the same).

@sean-rose what does the above mean? What do yo mean by the v2 API in app_name?

Leaving a change request here to mark that this should not be merged until Jan-Erik reviews.

**Review by badboy — COMMENTED** [review]
On top of Alessio's questions:

Is there any downstream tool that caches the API result and can potentially break now? (Dictionary?)

Otherwise, if this doesn't affect the raw ping table names this has my r+

File: repositories.yaml
Comment (brizental) [reply repositories.yaml:808]: Hey Sean, I am acutally working on the VPN team now so yeah, you can leave me there :D Thanks for asking!

**Comment by sean-rose** [comment]
> @Dexterp37 
> Will this change automatically generate an unified view?

No, I don't believe it will, but it indicates the intent that this is one set of telemetry, and maybe once [bug 1712332](https://bugzilla.mozilla.org/show_bug.cgi?id=1712332) "_Create UNIONed views for Glean app-datasets across all channels_" is resolved such unified views will be automatically generated.

In the meantime I plan to configure SQL generation of unified views/ETL for event funnels (like [was done for Fenix](https://bugzilla.mozilla.org/show_bug.cgi?id=1731093)), as well as manually creating some unified views.

> > Based on a dry run, the only consequential result of this change is in the v2 API the Android app's `app_name` would change from `mozilla_vpn_android` to `mozilla_vpn` (its `app_id` would remain the same).
> 
> @sean-rose what does the above mean? What do yo mean by the v2 API in app_name?

My understanding is probe scraper outputs two sets of JSON files that serve as its API:
- Files in the top-level `/glean` folder are what can be considered the original/v1 API, and this PR will only cause inconsequential changes there (descriptions and notification email lists).
- Files in the top-level `/v2` folder are the v2 API, and this PR will cause one consequential change in `/v2/glean/app-listings` (as previously described):
  ```diff
     {
       "app_channel": "release",
       "app_description": "Mozilla VPN is a VPN client application. The first\nMozilla premium service.\n",
       "app_id": "org.mozilla.firefox.vpn",
  -    "app_name": "mozilla_vpn_android",
  +    "app_name": "mozilla_vpn",
       "bq_dataset_family": "org_mozilla_firefox_vpn",
       "branch": "main",
  -    "canonical_app_name": "Mozilla VPN (Android)",
  +    "canonical_app_name": "Mozilla VPN",
       "dependencies": [
         "glean-core"
       ],
  +    "description": "Mozilla VPN (Android)",
       "document_namespace": "org-mozilla-firefox-vpn",
       "metrics_files": [
         "glean/metrics.yaml"
       ],
       ...
     }
  ```

File: repositories.yaml
Comment (sean-rose) [reply repositories.yaml:826]: Not as far as I can tell, but this part of our data platform is still new to me so I may simply be unaware.

Two things I've noticed:
- Fenix has multiple nightly channels, which would indicate multiple channels of the same type are generally OK.
- Some [Glean SQL generation code](https://github.com/mozilla/bigquery-etl/blob/c06700fc19762fbd3b9fc9e576bba73d22d13998/sql_generators/glean_usage/glean_app_ping_views.py#L52-L67) does look for a single release channel, but oddly it just [assumes the very first one is the release channel](https://github.com/mozilla/bigquery-etl/blob/c06700fc19762fbd3b9fc9e576bba73d22d13998/sql_generators/glean_usage/glean_app_ping_views.py#L56) (and doesn't look at the channel values).

**Comment by sean-rose** [comment]
In doing a test run of bigquery-etl SQL generation with the modified probe scraper output from this PR I discovered this would have the adverse affect of no longer generating `metrics_clients_daily` from VPN Android data because the main VPN client doesn't have a `metrics` table (does Glean.js not do metrics pings?).

Given that issue, and the fact that this PR would currently have no practical beneficial impacts, I'm inclined to close this PR without merging, and instead proceed with just a manual ETL solution (mozilla/bigquery-etl#3140).

We could revisit this scenario if/when [bug 1712332](https://bugzilla.mozilla.org/show_bug.cgi?id=1712332) "_Create UNIONed views for Glean app-datasets across all channels_" is resolved.

**Comment by badboy** [comment]
> > @badboy
> > Is there any downstream tool that caches the API result and can potentially break now? (Dictionary?)
> 
> Good question. Glean dictionary currently shows those as [separate](https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn) [apps](https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn_android). Any idea who would be best to ask about how Glean dictionary will react to this change?

@Iinh 

> 
> > Otherwise, if this doesn't affect the raw ping table names this has my r+
> 
> It shouldn't affect the raw ping tables, as those are based on the `app_id` rather than the `app_name`.

👍

**Comment by badboy** [comment]
> In doing a test run of bigquery-etl SQL generation with the modified probe scraper output from this PR I discovered this would have the adverse affect of no longer generating `metrics_clients_daily` from VPN Android data because the main VPN client doesn't have a `metrics` table (does Glean.js not do metrics pings?).

Correct. Glean.js does not have a `metrics` ping at the moment. That was one of the changes when VPN switched to Glean Kotlin.

> Given that issue, and the fact that this PR would currently have no practical beneficial impacts, I'm inclined to close this PR without merging, and instead proceed with just a manual ETL solution (https://github.com/mozilla/bigquery-etl/pull/3140).

I agree with that. Let's still wait for @linh to answer the question about the Dictionary, so we know it for next time.

**Comment by Iinh** [comment]
> Good question. Glean dictionary currently shows those as [separate](https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn) [apps](https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn_android). Any idea who would be best to ask about how Glean dictionary will react to this change?

This should not affect Glean Dictionary. With this change VPN would instead be shown as one product, with 2 different application IDs -- `mozillavpn` and `mozilla-vpn-android` -- under the App ID tab. The metric list will include metrics of both channels, with the colored labels attached showing the library origin so we should be able to tell which one is from gleanjs, glean-core, etc.

**Pull request closed without merging**
